File: easy/theleadgame/theleadgame.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# started at: Sunday, 24. November 2013 10:22PM
import sys


iterations = int(raw_input())

lead = {
    1: 0,
    2: 0
}

score1 = 0
score2 = 0

for i in xrange(iterations):
    scores = map(int,sys.stdin.readline().split())
    score1 += scores[0]
    score2 += scores[1]

    new_lead = score1-score2

    if new_lead < 0:
        if lead[2] < abs(new_lead):
            lead[2] = abs(new_lead)
    elif new_lead>0:
        if lead[1] < abs(new_lead):
           lead[1] = abs(new_lead)
    
if lead[1] < lead[2]:
    print '%s %s' % (2,lead[2])
else:
    print '%s %s' % (1,lead[1]) 

File: easy/theleadgame/theleadgame.c
/*
    started at: Sunday, 24. November 2013 10:22PM
*/

#include <stdio.h>
#include <stdlib.h>  // to suppress warning on using abs

int main(int argc, char const *argv[])
{
    int iterations = 0;
    scanf("%d",&iterations);

    int lead1 = 0;
    int lead2 = 0;
    int score1 = 0;
    int score2 = 0;
    int new_lead = 0;
    int temp = 0;

    for (int i = 0; i < iterations; ++i)
    {   
        scanf("%d",&temp);
        score1 += temp;
        scanf("%d",&temp);
        score2 += temp;

        new_lead = score1 - score2;

        if (new_lead<0)
        {
            if (lead2<abs(new_lead))
            {
                lead2 = abs(new_lead);
            }
        }
        else if (new_lead>0)
        {
            if (lead1<abs(new_lead))
            {
                lead1 = abs(new_lead);
            }
        }            
    }

    if (lead1<lead2)
    {
        printf("%d %d\n",2,lead2);
    }
    else
    {
        printf("%d %d\n",1,lead1);
    }

    return 0;
}